openhtf/util/files.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Helper functions for downloading and working with files."""
+
+import logging
+import os
+import shutil
+import zipfile
+
+try:  # for 2/3 compatibility
+  import urllib2 as urllib  # pylint: disable=g-import-not-at-top
+except NameError:
+  import urllib.request as urllib  # pylint: disable=g-import-not-at-top
+
+_LOG = logging.getLogger(__name__)
+
+
+def download(url, fd):
+  """Downloads a URL with HTTP to a file descriptor."""
+  try:
+    request = urllib.urlopen(url)  # Python 2 does not support with blocks
+    shutil.copyfileobj(request, fd)
+  finally:
+    try:
+      request.close()
+    except NameError:
```

Comment:
I think you can pull the request = assignment out of the try block and then you don't need to handle the NameError, as it won't hit the finally: block unless request has been successfully assigned to.